packages/labs/compiler/src/lib/ast-fragments.ts

```diff
@@ -105,22 +266,47 @@ export const createCompiledTemplateResult = ({
  * As an example output, a single BooleanAttributePart looks something like:
  *
  * ```ts
- * [{ type: 1, index: 0, name: "data-attr", strings: ["", ""], ctor: _$LH_BooleanAttributePart }]
+ * [{ type: 1, index: 0, name: "data-attr", strings: ["", ""], ctor: _B }]
  * ```
  */
 export const createTemplateParts = ({
   f,
   parts,
+  attributePartConstructorNameMap,
 }: {
   f: ts.NodeFactory;
   parts: TemplatePart[];
+  attributePartConstructorNameMap: AttributePartConstructorAliases;
 }) =>
   f.createArrayLiteralExpression(
     parts.map((part) => {
       const partProperties = [
         f.createPropertyAssignment('type', f.createNumericLiteral(part.type)),
         f.createPropertyAssignment('index', f.createNumericLiteral(part.index)),
       ];
+      if (part.type === PartType.ATTRIBUTE) {
+        const ctorAlias =
+          attributePartConstructorNameMap[
+            attributePartConstructors[
+              part.ctorType as keyof typeof attributePartConstructors
```

Comment:
The AttributeKind type should remove the need for this cast

Reply:
\*chefs kiss\*
Thank you!